db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java

```diff
@@ -0,0 +1,141 @@
+    protected int checkpointAfterMs;
+    protected boolean extraStatistics;
+    protected boolean resolveLinks;
+    protected boolean fromStart;
+    protected boolean fromEnd;
+    protected int historyBufferSize;
+    protected int liveBufferSize;
+    protected int maxCheckpointCount;
+    protected int maxRetryCount;
+    protected int maxSubscriberCount;
+    protected int messageTimeoutMs;
+    protected int minCheckpointCount;
+    protected int readBatchSize;
+    protected ConsumerStrategy strategy;
+
+    public AbstractPersistentSubscriptionSettingsBuilder() {
+        checkpointAfterMs = 2_000;
+        resolveLinks = false;
+        extraStatistics = false;
+        fromStart = false;
+        fromEnd = false;
+        messageTimeoutMs = 30_000;
+        maxRetryCount = 10;
+        minCheckpointCount = 10;
+        maxCheckpointCount = 1_000;
+        maxSubscriberCount = 0;
+        liveBufferSize = 500;
+        readBatchSize = 20;
+        historyBufferSize = 500;
+        strategy = ConsumerStrategy.RoundRobin;
+    }
+
+    public AbstractPersistentSubscriptionSettingsBuilder(AbstractPersistentSubscriptionSettings settings) {
+        checkpointAfterMs = settings.getCheckpointAfterMs();
+        resolveLinks = settings.isResolveLinks();
+        extraStatistics = settings.isExtraStatistics();
+        fromStart = settings.getFromStart();
+        fromEnd = settings.getFromEnd();
+        messageTimeoutMs = settings.getMessageTimeoutMs();
+        maxRetryCount = settings.getMaxRetryCount();
+        minCheckpointCount = settings.getMinCheckpointCount();
+        maxCheckpointCount = settings.getMaxCheckpointCount();
+        maxSubscriberCount = settings.getMaxSubscriberCount();
+        liveBufferSize = settings.getLiveBufferSize();
+        readBatchSize = settings.getReadBatchSize();
+        historyBufferSize = settings.getHistoryBufferSize();
+        strategy = settings.getStrategy();
+    }
+
+    public T enableLinkResolution() {
+        return resolveLinks(true);
+    }
+
+    public T disableLinkResolution() {
+        return resolveLinks(false);
+    }
+
+    public T resolveLinks(boolean value) {
+        this.resolveLinks = value;
+        return (T) this;
+    }
+
+    public T enableExtraStatistics() {
+        return extraStatistics(true);
+    }
+
+    public T disableExtraStatistics() {
+        return extraStatistics(false);
+    }
+
+    public T extraStatistics(boolean value) {
+        this.extraStatistics = value;
+        return (T) this;
+    }
+
+    public T checkpointAfterInMs(int value) {
+        this.checkpointAfterMs = value;
+        return (T) this;
+    }
+
+    public T fromStart() {
+        this.fromStart = true;
+        this.fromEnd = false;
+        return (T) this;
+    }
+
+    public T fromEnd() {
+        this.fromEnd = true;
+        this.fromStart = false;
+        return (T) this;
+    }
+
+
+    public T historyBufferSize(int value) {
```

Comment:
**SUGGESTION:** I think that typically in Java, such setters should start with the prefix `set` or the name should be an imperative form.

Reply:
Agree, but in this case it's a builder which normally has a more fluent syntax?

Reply:
Agree, but this is a builder class which normally has a more fluent naming?

Reply:
It is not the hill I'd die for, but IMHO, having set in this case would be more explicit in this case. But it's not a blocker for me to keep it as it is.